### PR TITLE
add DISC attentionF custom operator schema

### PR DESCRIPTION
**Pull request by Yancey1989**
This PR added a PyTorch custom operator `disc::attentionF` op schema, that we can use pdll to lower this op to disc BladNN custom call .

### CODE REVIEW

File: pytorch_blade/pytorch_blade/disc_ops/attention_op.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    bool is_causal,
+    bool compute_logsumexp) {
+  TORCH_CHECK(
+      false,
+      "nerver call this disc attentionF operator on eager mode, we need to lower this op to DISC custom call, please contact BladeDISC dev team!");
+  Tensor out, logsumexp;
+  return std::make_tuple(out, logsumexp);
+}
+
+TORCH_LIBRARY(disc, m) {
```
Comment (tanyokwok) [inline pytorch_blade/pytorch_blade/disc_ops/attention_op.cpp:33]: Nit: how about changing the module name from disc to torch_blade here?

Comment (Yancey1989) [reply pytorch_blade/pytorch_blade/disc_ops/attention_op.cpp:33]: `torch_blade` namespace has been used in `quantization` custom op, and the same namespace should be implemented in the same module. But I agree with you, using `torch_blade` or `disc` namespace for all custom operators is better, how about refacting them into one module in the next PR?

File: pytorch_blade/pytorch_blade/disc_ops/attention_op.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <torch/script.h>
+#include <cstdint>
+
+namespace torch {
+namespace blade {
+std::tuple<at::Tensor, at::Tensor> torch_blade_attentionF(
```
Comment (linbinskn) [inline pytorch_blade/pytorch_blade/disc_ops/attention_op.cpp:17]: Only single return value is needed for inference, is current api compatible with it? e.g. the second return value is None

File: pytorch_blade/pytorch_blade/disc_ops/attention_op.cpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  TORCH_CHECK(
+      false,
+      "nerver call this disc attentionF operator on eager mode, we need to lower this op to DISC custom call, please contact BladeDISC dev team!");
+  Tensor out, logsumexp;
+  return std::make_tuple(out, logsumexp);
+}
+
+TORCH_LIBRARY(disc, m) {
+  m.def(
+      "attentionF(Tensor query, Tensor key, Tensor value, Tensor attn_mask, str i_layout=\"BMHK\", str o_layout=\"BMHK\", bool is_causal=False, bool compute_logsumexp=False) -> (Tensor, Tensor)");
```
Comment (linbinskn) [inline pytorch_blade/pytorch_blade/disc_ops/attention_op.cpp:35]: Nit: add comment to clarify the arguments

File: pytorch_blade/pytorch_blade/disc_ops/attention_op.cpp (hunk 2 above)
Comment (Yancey1989) [reply pytorch_blade/pytorch_blade/disc_ops/attention_op.cpp:17]: let's return a none tensor if `compute_logsumexp` is false.

**Review by qiuxiafei — APPROVED** [review]

**Pull request merged**